ONSdigital/babbage
# Delete caching and metrics

### What

The Jira ticket is here: https://jira.ons.gov.uk/browse/DIS-923

The changes in this PR are to decommission the caching behaviour in babbage.

Importantly - we cannot deprecate anything around the storing of the ‘cache times’ as these are also used for legacy reindex. We are now storing cache times in two places:

documentdb (via dp-legacy-cache-api)
elasticsearch 2.4.2 (via babbage)

Babbage uses these cache times to work out what content needs reindexing for search - so if we remove them from ES2.4.2, legacy reindex will fail. Until we are able to migrate all search pages from babbage, these will need to stay.

### How to review

Please check that the following changes have been made:

- Remove feature flags put in: ENABLE_LEGACY_CACHE_API and the code now hidden behind it (i.e. remove code where it would trigger if this is 'N')
- Remove cache time calculation logic (largely within ContentClient.java)
- Remove feature flags for ENABLE_CACHE as we should now always be cache disabled from Babbage’s perspective. (i.e. remove code where it would trigger if set to 'Y')
- Remove ENABLE_METRICS feature flag, and the code behind it, as this is redundant without the ENABLE_CACHE functionality.
- Remove METRICS_FORMAT, POST_PUBLISH_CACHE_MAX_AGE and POST_PUBLISH_CACHE_EXPIRY_OFFSET as these are no longer needed either.

### Who can review

Anyone.
